app/frontend/wallet/account.ts

```diff
@@ -225,91 +222,68 @@ const Account = ({
+  async function signTxAux(txAux: _TxAux) {
     const signedTx = await cryptoProvider
-      .signTx(txAux, [], myAddresses.fixedPathMapper())
+      .signTx(txAux, myAddresses.fixedPathMapper())
       .catch((e) => {
         throw NamedError('TransactionRejectedWhileSigning', {message: e.message})
       })
     return signedTx
   }
 
-  async function getMaxSendableAmount(address, hasDonation, donationAmount, donationType) {
+  async function getMaxSendableAmount(address: _Address, sendAmount: SendAmount) {
     // TODO: why do we need hasDonation?
     const utxos = (await getUtxos()).filter(isUtxoProfitable)
-    return _getMaxSendableAmount(utxos, address, hasDonation, donationAmount, donationType)
+    return _getMaxSendableAmount(utxos, address, sendAmount)
   }
 
-  async function getMaxDonationAmount(address: string, sendAmount: Lovelace) {
-    const utxos = (await getUtxos()).filter(isUtxoProfitable)
-    return _getMaxDonationAmount(utxos, address, sendAmount)
-  }
-
-  async function getMaxNonStakingAmount(address) {
+  async function getMaxNonStakingAmount(address: _Address, sendAmount: SendAmount) {
     const utxos = (await getUtxos()).filter(({address}) => !isBase(addressToHex(address)))
-    return _getMaxSendableAmount(utxos, address, false, 0, false)
-  }
-
-  type utxoArgs = {
-    address?: string
-    donationAmount?: Lovelace
-    coins?: Lovelace
-    poolHash?: string
-    stakingKeyRegistered?: boolean
-    txType?: TxType
-    rewards: any
+    return _getMaxSendableAmount(utxos, address, sendAmount)
   }
 
-  const getTxPlan = async (args: utxoArgs) => {
-    const stakingAddress = await myAddresses.getStakingAddress()
-    const {address, coins, donationAmount, poolHash, stakingKeyRegistered, txType, rewards} = args
-    const changeAddress = await getChangeAddress()
+  const selectUTxOs = async (txType: TxType): Promise<UTxO[]> => {
     const availableUtxos = await getUtxos()
     const nonStakingUtxos = availableUtxos.filter(({address}) => !isBase(addressToHex(address)))
     const baseAddressUtxos = availableUtxos.filter(({address}) => isBase(addressToHex(address)))
     const randomGenerator = PseudoRandom(seeds.randomInputSeed)
-    // we shuffle non-staking utxos separately since we want them to be spend first
-    const shuffledUtxos =
-      txType === TxType.CONVERT_LEGACY
-        ? shuffleArray(nonStakingUtxos, randomGenerator)
-        : [
-          ...shuffleArray(nonStakingUtxos, randomGenerator),
-          ...shuffleArray(baseAddressUtxos, randomGenerator),
-        ]
-    const plan = selectMinimalTxPlan(
-      shuffledUtxos,
-      address,
-      coins,
-      donationAmount,
-      changeAddress,
-      stakingAddress,
-      poolHash,
-      !stakingKeyRegistered,
-      rewards
-    )
-    return plan
+    // TODO: ideally we want to do nonStaking, with token, only ada, the rest
+    if (txType === TxType.SEND_ADA || txType === TxType.CONVERT_LEGACY) {
```

Comment:
what's the point in having this condition? Is it addressing some edge case that would otherwise result in a suboptimal tx plan? Probably worth elaborating with a comment if it isn't actually redundant